### PR TITLE
Play 2.5

### PR DESCRIPTION
**Pull request by davidfurey**
Changes to support Play 2.5:
- ```play.api.mvc.Filter``` requires an ```akka.stream.Materializer```
- Dependency injection

### CODE REVIEW

**Comment by alexduf** [comment]
Looks good to me

File: example/build.sbt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name := "play-googleauth-example"
 
-version := "0.3.5-SNAPSHOT"
+version := "0.3.8-SNAPSHOT"
```
Comment (piuccio) [inline example/build.sbt]: should the new version be `0.4` or is this backward compatible?

Comment (cb372) [inline example/build.sbt]: I agree, we should bump it to 0.4, as bumping Play to 2.5.x is a breaking change.

The usual way of handling versioning in cases like this is to increment your library's version whenever you increment the corresponding Play version. e.g. 0.3.x is compatible with Play 2.4.x, 0.4.x is compatible with Play 2.5.x, etc. If we ever need to release a fix for the Play 2.4.x version of the lib, we can create a `0.3.x` branch from the `v0.3.7` tag.

I'd also add a note to the readme explaining this versioning policy.

**Comment by cb372** [comment]
:+1: apart from the comment about versioning

**Comment by mchv** [comment]
@davidfurey any way to avoid dependency inhjection?

**Comment by davidfurey** [comment]
The dependency injection is only in the example project.  I think it might be possible to still use a static router in Play 2.5 but it is not recommended.  @alexduf has discovered https://github.com/adamw/macwire which is making compile time dependency injection quite easy.

**Comment by mchv** [comment]
@davidfurey this recommendation is a [madness]((http://stackoverflow.com/questions/5192904/play-framework-uses-a-lot-of-statics/5193721#5193721)) [from new playframework developers](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/play-framework/hGP-6fsKUYk/4jRvzUT5GQAJ). Dependency injection is generally useless and should be generally avoided.

**Pull request merged**